### PR TITLE
Google Spreadsheet to JSON converter

### PR DESCRIPTION
**Pull request by rbalekai**

### CODE REVIEW

File: data/administrative_units.json [hunk 1]
```diff
@@ -609,6 +601,14 @@
   },
   {
     "predicate_name": "administrative_units",
+    "term_label": "University of Notre Dame::College of Arts and Letters::John J. Reilly Center for Science, Technology and Values",
```
Comment (jeremyf) [inline data/administrative_units.json]: It appears as though you committed changes from running the Google Sheets to JSON converter as well as the code for the converter. Is that correct?

Comment (rbalekai) [inline data/administrative_units.json]: Right. Do you want that to be removed from the commit?

File: script/google_spreadsheet.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  def convert_to_json(data)
+    json_array = []
+    data.each do |key, value|
+      json_string = Jbuilder.encode do |json|
+        json.set!("predicate_name", "administrative_units")
+        json.set!("term_label", key)
+        json.set!("term_uri", nil)
+        json.set!("default_presentation_sequence", ORDER[value])
+        json.set!("activated_on", "2015-07-22")
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: Why was this date chosen? Should it be today? How does this relate to data that comes and goes from the Google Sheet?

Comment (rbalekai) [inline script/google_spreadsheet.rb]: Yeah. That was a test which got retained. I will change that

File: script/google_spreadsheet.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      final[ line.join('::') ] = row[3]
+      line = []
+    end
+    final
+  end
+
+  def convert_to_json(data)
+    json_array = []
+    data.each do |key, value|
+      json_string = Jbuilder.encode do |json|
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: I don't believe JBuilder is necessary. If I'm understanding what you are doing you are:

Writing JSON String (line 72) -> Parsing JSON String to Hash (line 80) -> Writing Hash to JSON String (line 87)

I believe JSON.pretty_generate takes a Ruby object and makes a JSON object. You have one additional step.

File: script/google_spreadsheet.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        json.set!("activated_on", "2015-07-22")
+        json.set!("deactivated_on", nil)
+      end
+      json_array << JSON.parse(json_string)
+    end
+    json_array
+  end
+
+  def write_to_file(data)
+    f = File.new(output_filepath, "w")
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: Don't use File.new; Instead use `File.open(output_filepath, 'w+') { |file| ... }`

If you use File.new and encounter an exception on line 87, then 88 and 89 don't execute. The block ensures that flush and close are performed.

File: script/google_spreadsheet.rb [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  def client_secrets
+    return @secrets if @secrets
+    secrets_path = File.join(File.dirname(FILE_PATH), '../config/client_secrets.example.yml')
+    secrets_path = File.join(File.dirname(FILE_PATH), '../config/client_secrets.yml') if File.exist? File.join(File.dirname(FILE_PATH), '../config/client_secrets.yml')
+    @secrets = YAML.load(File.open(File.join(secrets_path)))
+  end
+
+end
+spread_sheet = GoogleSpreadsheet.new(document_key: ARGV[0], output_filepath: ARGV[1])
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: ARGV[0] looks rather magical as does ARGV[1]. Is this documented? What about using an ENV variable?

File: script/google_spreadsheet.rb [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        row_data << ws[row, col]
+      end
+      @spreadsheet_data << row_data
+    end
+  end
+
+  def get_required_data_from_spreadsheet
+    final = {}
+    line = []
+    spreadsheet_data.shift
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: What is happening here? I assume you are discarding the header lines. What happens if columns are rearranged?

File: script/google_spreadsheet.rb [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      (1..ws.num_cols).each do |col|
+        row_data << ws[row, col]
+      end
+      @spreadsheet_data << row_data
+    end
+  end
+
+  def get_required_data_from_spreadsheet
+    final = {}
+    line = []
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: Why is line declared here? I was expecting `line=[]` to be between line 59 and line 60.

File: script/google_spreadsheet.rb [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    spreadsheet_data.shift
+    spreadsheet_data.each_with_index do |row, index|
+      line << row[0]
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: What are we expecting at `row[0]`, `row[1]`, `row[2]`?

File: script/google_spreadsheet.rb [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      line << row[1] if !row[1].empty?
+      line << row[2] if !row[2].empty?
+      final[ line.join('::') ] = row[3]
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: What is `row[3]`?

Perhaps some constants for `0, 1, 2, 3`

File: script/google_spreadsheet.rb (hunk 3 above)
Comment (jeremyf) [inline script/google_spreadsheet.rb]:
```ruby
{ "predicate_name" => "administrative_units" }
```

File: script/google_spreadsheet.rb [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+      line = []
+    end
+    final
+  end
+
+  def convert_to_json(data)
+    json_array = []
+    data.each do |key, value|
+      json_string = Jbuilder.encode do |json|
+        json.set!("predicate_name", "administrative_units")
```
Comment (jeremyf) [inline script/google_spreadsheet.rb]: How do we know this administrative_units? Is it related to ARGV[1]? ARGV[0]?

File: data/administrative_units.json (hunk 1 above)
Comment (jeremyf) [inline data/administrative_units.json]: It can stay

**Pull request closed without merging**